src/widgets/table/table_state.rs

```diff
@@ -49,6 +53,7 @@
 pub struct TableState {
     pub(crate) offset: usize,
     pub(crate) selected: Option<usize>,
+    pub(crate) selected_column: Option<usize>,
```

Comment:
One thing to keep in mind is how would this new field play out if we decide to consolidate the functionalities of `TableState` and `ListState` in the future. Hint: #1215

Reply:
Yes, I already saw that issue. It is also one of the reasons I decided to leave the selection methods as is for now (instead of renaming the methods `select_row` and so on). Though I am not sure what I should change to avoid a problem. I don't think this is a problem for the consolidation as we will still have to store an `Option<usize>` somewhere in the table state. If you have any ideas on what should be done to make this more compatible, I am happy to make the changes. I just don't know what should be changed here.

Reply:
Yeah, I'm not sure what to change here as well. It should be fine. I just wanted to bring that up as a point related to `TableState`. I would very much like to avoid a lot of changes between these two states because I believe they should be consolidated in the future.